lib/src/main/kotlin/dev/chrisbanes/snapper/SnapperFlingBehavior.kt

```diff
@@ -209,20 +257,60 @@ public object SnapOffsets {
  *
  * Note: the default parameter value for [decayAnimationSpec] is different to the value used in
  * [rememberSnapperFlingBehavior], due to not being able to access composable functions.
- *
- * @param layoutInfo The [SnapperLayoutInfo] to use.
- * @param decayAnimationSpec The decay animation spec to use for decayed flings.
- * @param springAnimationSpec The animation spec to use when snapping.
- * @param maximumFlingDistance Block which returns the maximum fling distance in pixels.
- * The returned value should be > 0.
  */
 @ExperimentalSnapperApi
-public class SnapperFlingBehavior(
+public class SnapperFlingBehavior private constructor(
     private val layoutInfo: SnapperLayoutInfo,
-    private val maximumFlingDistance: (SnapperLayoutInfo) -> Float = SnapperFlingBehaviorDefaults.MaximumFlingDistance,
     private val decayAnimationSpec: DecayAnimationSpec<Float>,
-    private val springAnimationSpec: AnimationSpec<Float> = SnapperFlingBehaviorDefaults.SpringAnimationSpec,
+    private val springAnimationSpec: AnimationSpec<Float>,
+    private val snapIndex: (SnapperLayoutInfo, targetIndex: Int) -> Int,
+    private val maximumFlingDistance: (SnapperLayoutInfo) -> Float,
 ) : FlingBehavior {
+    /**
+     * @param layoutInfo The [SnapperLayoutInfo] to use.
+     * @param decayAnimationSpec The decay animation spec to use for decayed flings.
+     * @param springAnimationSpec The animation spec to use when snapping.
+     * @param snapIndex Block which returns the index to snap to. The block is provided with the
+     * [SnapperLayoutInfo] and the index which Snapper has determined is the correct target index.
+     * Callers can override this value to any valid index for the layout. Some common use cases include
+     * limiting the fling distance, and rounding up/down to achieve snapping to groups of items.
+     */
+    public constructor(
```

Comment:
idk, but I succeed in using both maxFlingDistance and snapIndex at the same time. 
It'll not be supported by snapper?

Reply:
It is theoretically possible to want to have both, but they kinda conflict conceptually. why did you use both of them?

Reply:
snapIndex to snap by group of 3/4 elements, flingDistance to reduce fling/scroll distance. 
Indeed I'm not sure 2nd will be needed, because I'm only trying api, not checking in real project

Reply:
> When I'm scrolling by the 1.6+ items, I want to snap it to the next group (so I need sort of predicted offset after decay finished). When I scrolled to 1.6- items, I want to snap it to the start index of current group.
> It looks, like I can't solve it without additional param in snapIndex.
> Could you, please, advice something?

We don't supply the offset on purpose, because we don't know what it will be (the items won't be laid out). The best we can do is a predicted index.

Reply:
Also note, we always round the determined index, so flinging to `1.6` will provide a value of `2`. What you do with that value is your decision though.

Going to mark this as resolved as I don't think there's anything to do here. Feel free to comment if I'm missing something.